dubbo-remoting/dubbo-remoting-zookeeper/src/main/java/org/apache/dubbo/remoting/zookeeper/curator/CuratorZookeeperClient.java

```diff
@@ -354,10 +365,24 @@ public void nodeChanged() throws Exception {
 
     static class CuratorWatcherImpl implements CuratorWatcher {
 
+        private static volatile ExecutorService CURATOR_WATCHER_EXECUTOR_SERVICE;
+
+        private static volatile boolean closed = false;
```

Comment:
What is this state used for?

Reply:
if there is an event callback after closing, the ``null thread pool`` will be reinitialized.

Reply:
just check if CURATOR_WATCHER_EXECUTOR_SERVICE is null is enough?

Reply:
if only check CURATOR_WATCHER_EXECUTOR_SERVICE is null, it will be initialized when Curator is closed and event callbacks are executed concurrently